sys/x86/isa/atpic.c

```diff
@@ -526,6 +527,9 @@ atpic_handle_intr(u_int vector, struct trapframe *frame)
 {
 	struct intsrc *isrc;
 
+	critical_enter();
+	++curthread->td_intr_nesting_level;
+
 	kasan_mark(frame, sizeof(*frame), sizeof(*frame), 0);
 	kmsan_mark(frame, sizeof(*frame), KMSAN_STATE_INITED);
 	trap_check_kstack();
```

Comment:
I'm unsure of the ordering here.  Should these instead be in the opposite order?  (similar issue below in `local_apic.c`)